sycl/include/sycl/accessor.hpp

```diff
@@ -3087,6 +3293,27 @@ class host_accessor
                       PropertyList, CodeLoc) {}
 
 #endif
+
+  /// Compares the host_accessor against a weak object using an owner-based
+  /// implementation-defined ordering.
+  ///
+  /// \param Other is the weak object to compare ordering against.
+  /// \return true if this object precedes \param Other and false otherwise.
+  bool ext_oneapi_owner_before(
+      const ext::oneapi::detail::weak_object_base<host_accessor> &Other)
+      const noexcept {
+    return detail::AccessorBaseHost::impl.owner_before(
+        ext::oneapi::detail::getSyclWeakObjImpl(Other));
+  }
+
+  /// Compares the host_accessor against another host_accessor using an
+  /// owner-based implementation-defined ordering.
+  ///
+  /// \param Other is the object to compare ordering against.
+  /// \return true if this object precedes \param Other and false otherwise.
+  bool ext_oneapi_owner_before(const host_accessor &Other) const noexcept {
+    return detail::AccessorBaseHost::impl.owner_before(Other.impl);
+  }
```

Comment:
Can we use CRTP to implement it just once and then enable in final classes via inheritance?

Reply:
Interesting idea! I've tried implementing it. Is this roughly what you had in mind?

Reply:
As per offline discussion we agreed that the CRTP base class could be applied to all classes defining the interface rather than just accessors.